### PR TITLE
app-misc/chkcrontab: migrate to PEP517 build

### PR DESCRIPTION
**Pull request by hydrapolic**
Tested on ~amd64, thanks!

```bash
--- chkcrontab-1.7-r2.ebuild    2023-01-25 14:11:28.149851271 +0000
+++ chkcrontab-1.7-r3.ebuild    2024-03-11 13:35:20.431449380 +0000
@@ -1,11 +1,10 @@
-# Copyright 1999-2023 Gentoo Authors
+# Copyright 1999-2024 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
 EAPI=8
 
-PYTHON_COMPAT=( python3_{9..11} )
-DISTUTILS_USE_SETUPTOOLS=no
-
+DISTUTILS_USE_PEP517=setuptools
+PYTHON_COMPAT=( python3_{10..12} )
 inherit distutils-r1
 
 DESCRIPTION="A tool to detect crontab errors"
@@ -14,11 +13,11 @@
 
 LICENSE="Apache-2.0"
 SLOT="0"
-KEYWORDS="amd64 x86"
-IUSE="test"
-RESTRICT="!test? ( test )"
+KEYWORDS="~amd64 ~x86"
+
+distutils_enable_tests pytest
 
-distutils_enable_tests setup.py
+PATCHES=( "${FILESDIR}/${PN}-man.patch" )
 
 python_install_all() {
        doman doc/${PN}.1
```

### CODE REVIEW

**Review by mgorny — CHANGES_REQUESTED** [review]

**Comment by hydrapolic** [comment]
Thank you for the review @mgorny, update pushed.

File: app-misc/chkcrontab/chkcrontab-1.7-r3.ebuild [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+DESCRIPTION="A tool to detect crontab errors"
+HOMEPAGE="https://github.com/lyda/chkcrontab"
+SRC_URI="https://github.com/lyda/chkcrontab/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="Apache-2.0"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
+RESTRICT="!test? ( test )"
```
Comment (mgorny) [inline app-misc/chkcrontab/chkcrontab-1.7-r3.ebuild:18]: The eclass takes care of that. Even if it didn't, you aren't actually using it anywhere.

Comment (hydrapolic) [reply app-misc/chkcrontab/chkcrontab-1.7-r3.ebuild]: Didn't know that, thanks for pointing that out. Updated the PR.

**Review by mgorny — CHANGES_REQUESTED** [review]
Tests fail with python3.12:

```pytb
FAILED tests/test_check.py::FSMUnitTest::testComplex - AttributeError: 'FSMUnitTest' object has no attribute 'assertEquals'. Did you me
an: 'assertEqual'?
```

and so on. You can `sed` them test methods but please send a proper fix upstream.

**Comment by hydrapolic** [comment]
> Tests fail with python3.12:
> 
> ```
> FAILED tests/test_check.py::FSMUnitTest::testComplex - AttributeError: 'FSMUnitTest' object has no attribute 'assertEquals'. Did you me
> an: 'assertEqual'?
> ```
> 
> and so on. You can `sed` them test methods but please send a proper fix upstream.

Seems like there is fork at https://github.com/gregorg/chkcrontab, but the tests fail there too. @hannob contributed some python 3.12 changes, I'll try to ask if the tests can be fixed.

**Comment by hydrapolic** [comment]
> Tests fail with python3.12:
> 
> ```
> FAILED tests/test_check.py::FSMUnitTest::testComplex - AttributeError: 'FSMUnitTest' object has no attribute 'assertEquals'. Did you me
> an: 'assertEqual'?
> ```
> 
> and so on. You can `sed` them test methods but please send a proper fix upstream.

Tests now pass on py3.12, pushed an update to the PR.

**Comment by mgorny** [comment]
Thanks!

**Comment by hydrapolic** [comment]
Thank you @mgorny

**Pull request closed without merging**